File: Facultate/Anul 2/Sem 2/MPP/Laburi/JavaTema4/JavaTema4/Shared/src/main/java/services/IMainService.java
package services;

import dtos.RideDTO;
import models.Reservation;
import models.Ride;
import utils.Observer;

import java.time.LocalDate;
import java.util.List;

public interface IMainService {

    public List<RideDTO> findRides(LocalDate rideDate, String departureHour, String destination);
    public List<Ride> getAllRides();
    public void makeReservation(Long rideId, String text, int noSeats);
    public void cancelReservation(Long reservationId);
    public Ride getRide(Long rideId);
    public List<Reservation> findReservations();
    void addObserver(Observer observer);

}

File: Facultate/Anul 2/Sem 2/MPP/Laburi/Edi/CSharpServer/Shared/services/IMainService.cs
using Shared.dtos;
using Shared.models;
using Shared.utils;

namespace Shared.services;

public interface IMainService
{
    List<RideDTO> FindRides(DateOnly rideDate, string departureHour, string destination);
    List<Ride> GetAllRides();
    void MakeReservation(long rideId, string clientName, int noSeats);
    void CancelReservation(long reservationId);
    Ride? GetRide(long rideId);
    List<Reservation> FindReservations();

    void AddObserver(IObserver observer);
    void RemoveObserver(IObserver observer);
}